application/src/test/java/org/togetherjava/tjbot/commands/tophelper/TopHelperAntiCheatTest.java

```diff
@@ -0,0 +1,71 @@
+package org.togetherjava.tjbot.commands.tophelper;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+final class TopHelperAntiCheatTest {
+
+    Predicate<String> uncountedCharsMatchPredicate =
+            TopHelpersMessageListener.UNCOUNTED_CHARS.asMatchPredicate();
+
+
+    @ParameterizedTest
+    @MethodSource("provideInvisibleNamedCharacters")
+    @DisplayName("Does exclude invisible characters")
+    void excludesInvisibleCharacters(char character, String name) {
+        String characterString = String.valueOf(character);
+
+        String errorMessage = "Character [%s] was unrightfully matched".formatted(name);
+
+        assertTrue(uncountedCharsMatchPredicate.test(characterString), errorMessage);
```

Comment:
Try to follow a GTW (given then when) pattern. makes it easier to understand the test:
```java
// GIVEN an invisible character
String invisibleCharText = String.valueOf(invisibleChar);

// WHEN checking whether its counted
boolean isNotCounted = isNotCounted.test(invisibleCharText);

// THEN expect that its not counted
assertTrue(isNotCounted, charName);
```

it is okay if you ommit the commits in simple cases like this, but still try to follow the pattern. so:
```java
String invisibleCharText = String.valueOf(invisibleChar);

boolean isNotCounted = isNotCounted.test(invisibleCharText);

assertTrue(isNotCounted, charName);
```

Reply:
I have to agree that makes it a lot more readable